### PR TITLE
Implement multiple tabs

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: ui/src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -22,12 +23,42 @@ type SidebarLinkProps = {
 
 const Sidebar = ({ location } : RouteComponentProps) => {
   var classes = useStyles();
+  const party = useParty();
+
+  var whatIsVisibleByWhom = new Map([
```
Comment (tamaskalcza-da) [inline ui/src/components/Sidebar/Sidebar.tsx:28]: Why do you use `var`s? It seems you can get by with `const` just fine. If not, using `let` would be more appropriate.

File: ui/src/components/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -22,12 +23,42 @@ type SidebarLinkProps = {
...
+  ]);
+  var panelNames = new Map([
+    ['buyerSellerRelationships', "Buyer Seller Relationships"],
+    ['quoteRequests', "Quote Requests"],
+    ['quoteRequestsAccepted', "Accepted Quote Requests"],
+  ]);
+
+  function SidebarItem(props: { identifier: string }) {
+    var partiesWhoCanSee = whatIsVisibleByWhom.get(props.identifier);
+    if (partiesWhoCanSee !== undefined && partiesWhoCanSee.includes(party)) {
```
Comment (tamaskalcza-da) [inline ui/src/components/Sidebar/Sidebar.tsx:43]:
```suggestion
    if (partiesWhoCanSee && partiesWhoCanSee.includes(party)) {
```

File: ui/src/components/Util.tsx [hunk 3]
```diff
@@ -26,8 +26,8 @@ export function shorten(text: any): any {
     if (isParty(text)) {
       return addSpacesBetweenWords(getDisplayName(text));
     }
-    if (text.length > 20) {
-      return `${text.substr(0,20)}...`;
+    if (text.length > 40) {
+      return `${text.substr(0,40)}...`;
```
Comment (tamaskalcza-da) [inline ui/src/components/Util.tsx:30]: Why `40` (or `20` for that matter...)? I'd consider getting this as a parameter. It likely makes more sense to specify this on a case-by-case basis.

File: ui/src/pages/quoteRequests/OrderedProductList.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+export function OrderedProductList(
+    props:
+      {
+        ledger: Ledger,
+        createEvent: CreateEvent<QuoteRequest, unknown, string> | undefined,
+        isDialogOpen: boolean,
+        setDialogOpen: React.Dispatch<React.SetStateAction<boolean>>
+      }) {
+    const fields = props.createEvent === undefined ? [] : props.createEvent.payload.products
```
Comment (tamaskalcza-da) [inline ui/src/pages/quoteRequests/OrderedProductList.tsx:20]:
```suggestion
    const fields = props.createEvent ? props.createEvent.payload.products : []
```

File: ui/src/pages/quoteRequests/OrderedProductList.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    return (
+      <Dialog open={props.isDialogOpen} key="quoteReq" onClose={() => ({})} maxWidth={false} fullWidth>
+      <DialogTitle>
+        Quote request {props.createEvent !== undefined ? `(Ordered by ${props.createEvent.payload.buyer})` : ""}
```
Comment (tamaskalcza-da) [inline ui/src/pages/quoteRequests/OrderedProductList.tsx:25]:
```suggestion
        Quote request {props.createEvent ? `(Ordered by ${props.createEvent.payload.buyer})` : ""}
```

File: ui/src/pages/quoteRequests/QuoteRequests.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+import { useState } from "react";
+import { CreateEvent } from "@daml/ledger";
+import { OrderedProductList } from "./OrderedProductList";
+
+export default function QuoteRequests() {
+
+  const party = useParty();
+  const ledger = useLedger();
+  const roles =
+    useStreamQuery(QuoteRequest);
```
Comment (tamaskalcza-da) [inline ui/src/pages/quoteRequests/QuoteRequests.tsx:19]: Extremely minor, but triggered my OCD. :) Ignore, if this is done by formatter.
```suggestion
  const roles = useStreamQuery(QuoteRequest);
```

File: ui/src/pages/quoteRequests/QuoteRequests.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            createEvent : CreateEvent<QuoteRequest>,
+            workflowId : string) {
+    ledger.exercise(
+      QuoteRequest.QuoteRequest_Accept,
+      createEvent.contractId,
+      { workflowId: workflowId });
+  };
+
+  const [isDialogOpen, setDialogOpen] = useState(false);
+  const [createEvent, setCreateEvent] = useState(undefined as CreateEvent<QuoteRequest> | undefined);
```
Comment (tamaskalcza-da) [inline ui/src/pages/quoteRequests/QuoteRequests.tsx:31]: Wouldn't this be equivalent?
```suggestion
  const [createEvent, setCreateEvent] = useState(undefined as CreateEvent<QuoteRequest>);
```

File: ui/src/pages/quoteRequestsAccepted/QuoteRequestsAccepted.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+import Contracts from "../../components/Contracts/Contracts";
+import { useStreamQuery, useParty, useLedger } from "@daml/react";
+import { QuoteRequest, QuoteRequestAccepted }
+  from "@daml.js/supplychain-1.0.0/lib/DA/RefApps/SupplyChain/QuoteRequest";
+import { useState } from "react";
+import { CreateEvent } from "@daml/ledger";
+import { OrderedProductList } from "../quoteRequests/OrderedProductList";
+import { SortedPartyNames } from "../login/Login";
+
+export default function QuoteRequestsAccepted() {
```
Comment (tamaskalcza-da) [inline ui/src/pages/quoteRequestsAccepted/QuoteRequestsAccepted.tsx:15]: What is the difference between this one and [QuoteRequests.tsx](ui/src/pages/quoteRequests/QuoteRequests.tsx)?

File: ui/src/pages/supplyInvitations/SupplyInvitations.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import { useState } from "react";
+import { CreateEvent } from "@daml/ledger";
+import { OrderedProductList } from "../quoteRequests/OrderedProductList";
+
+export default function SupplyInvitations() {
+
+  const party = useParty();
+  const ledger = useLedger();
+  const roles =
+    useStreamQuery(QuoteRequestSupplyInvitation);
```
Comment (tamaskalcza-da) [inline ui/src/pages/supplyInvitations/SupplyInvitations.tsx:19]: Again, ignore if this is done by a formatter.
```suggestion
  const roles = useStreamQuery(QuoteRequestSupplyInvitation);
```

File: ui/src/pages/quoteRequests/QuoteRequests.tsx (hunk 7 above)
Comment (gaborh-da) [reply ui/src/pages/quoteRequests/QuoteRequests.tsx:31]: No

File: ui/src/pages/quoteRequestsAccepted/QuoteRequestsAccepted.tsx (hunk 8 above)
Comment (gaborh-da) [reply ui/src/pages/quoteRequestsAccepted/QuoteRequestsAccepted.tsx:15]: One works with QuoteRequest-s, the other with QuoueRequestAccepted-s. Sadly, the naming is very bad (of the contracts).

**Pull request merged**